pageserver/src/tenant/timeline.rs

```diff
@@ -3323,7 +3323,6 @@ impl Timeline {
     ) -> Result<CompactLevel0Phase1Result, CompactionError> {
         let layers = self.layers.read().unwrap();
         let mut level0_deltas = layers.get_level0_deltas()?;
-        drop(layers);
 
```

Comment:
Note that we're in a `spawn_blocking` sync function here. So, we could just use `blocking_read`.

But, that wastes a spawn-blocking-thread-pool slot.

The patch that converts `Timeline::layers` to an async rwlock will move the locked `layers` into the `spawn_blocking`.